chunk.go

```diff
@@ -47,7 +47,10 @@ func decodeLargeChunk(r io.Reader, rowCount int, cellCount int) ([][]*string, er
 		return nil, err
 	}
 
-	return rows, nil
+	chunkRows := make([]chunkRowType, len(rows))
+	populateJSONRowSet(chunkRows, rows)
```

Comment:
hide the logic of making chunkRows inside populateJSONRowSet and make the signature of populateJSONRowSet as: chunkRows = populateJsonRowSet(rows), this way you have a clear input/output.